ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceItemView.kt

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.datacollection.tasks.multiplechoice
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.ClickableText
+import androidx.compose.material3.Checkbox
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.RadioButton
+import androidx.compose.material3.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.google.android.ground.ExcludeFromJacocoGeneratedReport
+import com.google.android.ground.R
+import com.google.android.ground.model.task.MultipleChoice
+import com.google.android.ground.model.task.Option
+import com.google.android.ground.ui.theme.AppTheme
+
+@Composable
+fun MultipleChoiceItemView(
+  item: MultipleChoiceItem,
+  modifier: Modifier = Modifier,
+  isLastIndex: Boolean = false,
+  toggleItem: (item: MultipleChoiceItem) -> Unit = {},
+  otherValueChanged: (text: String) -> Unit = {},
```

Comment:
how about have state and events

something like

```kt
val _multiplaeChoiceState = MutableStateFlow<MultiplaeChoiceData>(MultiplaeChoiceData())
```

```kt
val _multiplaeChoiceEvent = MutableSharedFlow<MultiplaeChoiceEvents>()
```

Reply:
Are there any benefits to using sharedFlow in this case?

Reply:
1. all events can be managed at a single place
2. can also prevent the double click
3. no multiple parameters to compose functions

Reply:
I tried that but the tests started failing after that. Can we handle that separately?